### DISCUSSION TITLE
REST API  -

### DISCUSSION THREAD

**Discussion by mrdeadlocked** [grafana/grafana-infinity-datasource#684] [Specific APIs]
I am trying to get data pulled in from SNIPE-IT's REST API. The data that comes in is one big row when I use the backend parser. I Can't get the rows/columns to work the same way as the default. What am I missing?

![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/461352/8d7392ce-7ba3-40e1-b1bc-569ac22fc1a6)

Example of a entry in the JSON
    {
        "age": "5 months ago",
        "alt_barcode": "http://192.168.1.84/uploads/barcodes/c128-1112787153.png",
        "asset_eol_date": null,
        "asset_tag": "1112787153",
        "assigned_to": null,
        "available_actions": {
            "checkin": true,
            "checkout": true,
            "clone": true,
            "delete": true,
            "restore": false,
            "update": true
        },
        "book_value": "601.15",
        "byod": false,
        "category": {
            "id": 5,
            "name": "Displays"
        },
        "checkin_counter": 0,
        "checkout_counter": 0,
        "company": null,
        "created_at": {
            "datetime": "2023-05-02 23:48:47",
            "formatted": "2023-05-02 11:48 PM"
        },
        "custom_fields": {
        },
        "deleted_at": null,
        "eol": 12,
        "expected_checkin": null,
        "id": 2560,
        "image": "http://192.168.1.84/uploads/models/ultrafine.jpg",
        "last_audit_date": null,
        "last_checkout": null,
        "location": {
            "id": 7,
            "name": "Kuhlmanport"
        },
        "manufacturer": {
            "id": 7,
            "name": "LG"
        },
        "model": {
            "id": 17,
            "name": "Ultrafine 4k"
        },
        "model_number": "4024007178360730",
        "name": "",
        "next_audit_date": null,
        "notes": "Created by DB seeder",
        "order_number": "33378247",
        "purchase_cost": "1,030.54",
        "purchase_date": {
            "date": "2023-04-02",
            "formatted": "2023-04-02"
        },
        "qr": null,
        "requests_counter": 0,
        "rtd_location": {
            "id": 7,
            "name": "Kuhlmanport"
        },
        "serial": "8555cff0-7831-3033-8337-ceebbc5052c2",
        "status_label": {
            "id": 1,
            "name": "Ready to Deploy",
            "status_meta": "deployable",
            "status_type": "deployable"
        },
        "supplier": {
            "id": 5,
            "name": "Schmitt, Schmeler and Johnston"
        },
        "updated_at": {
            "datetime": "2023-05-02 23:49:13",
            "formatted": "2023-05-02 11:49 PM"
        },
        "user_can_checkout": true,
        "warranty_expires": null,
        "warranty_months": null
    }

**Comment by yesoreyeram** [comment]
In the root selector specify `rows` as your array of data is inside that element

**Comment by mrdeadlocked** [comment]
Awesome. That did work. What would be the easiest way to transform the data to get more clear info. I've got something work with 'extract fields' and 'organize fields' Just curious if a better method might exist

![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/461352/2e577c46-b987-4389-aea1-72fcb026b609)

**Reply by yesoreyeram** [reply]
If you using uql, then use `project` command. If backend parser, just the fields should do the trick.

**Comment by mrdeadlocked** [comment]
I am using the backend parser so that i can try to make a public dashboard. Where would I add the fields in for the backend parser?

**Reply by yesoreyeram** [reply]
In the fields/columns section

**Reply by mrdeadlocked** [reply]
![image](https://github.com/yesoreyeram/grafana-infinity-datasource/assets/461352/70b5e359-f215-4371-a229-1b8848ac1a94)
I am able to get it to just show a column, but I still have the json data. What would be the best method to get that clean

**Reply by yesoreyeram** [reply]
`available_actions.checkin` as a column selector. Likewise, you need to specify the list of columns you want.

**Comment by mrdeadlocked** [comment]
Awesome. I got this to work. is it possible to change the record limit from 500 for a table view?
